api/controllers/EventController.js

```diff
@@ -89,16 +96,46 @@ var EventController = {
     },
 
     getEvents: function(req, res) {
+        var sortOption = req.swagger.params.sortBy.value;
         var page = req.swagger.params.page.value;
         var pageSize = req.swagger.params.pageSize.value;
 
-        Event.find({ 'hasFinished': false }, function (e, events){
-            if (e) return res.status(500).json(e.message);
-
-            res.json({events: events});
-        })
-        .skip((page-1)*pageSize)
-        .limit(pageSize)
+        var lng = req.swagger.params.lng.value;
+        var lat = req.swagger.params.lat.value;
+        var radius = req.swagger.params.radius.value;
+        
+        if(lat && lng){
```

Comment:
Since they are required params in the request so this `if` would always be true. What we really need is to validate `lat` and `lng`. if it's valid coordinate, then find the nearest events, or return events in chronological order

Reply:
cool, got it.